code/game/machinery/metaldetector.dm

```diff
@@ -0,0 +1,87 @@
+GLOBAL_LIST_INIT(metal_detector_items, typecacheof(list(
+	/obj/item/weapon/gun,
+	/obj/item/weapon/material,
+	/obj/item/weapon/melee,
+	/obj/item/device/transfer_valve,
+	/obj/item/weapon/grenade/,
+	/obj/item/ammo_casing/,
+	/obj/item/ammo_magazine
+	)))
+
+/obj/machinery/metal_detector
+	name = "metal detector"
+	desc = "An advanced metal detector used to detect weapons."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "metal_detector"
+	anchored = TRUE
+	req_one_access = list(access_security, access_heads)
+	idle_power_usage = 100
+
+	plane = MOB_PLANE
+	layer = ABOVE_MOB_LAYER
+
+	var/next_announcement_time	// so this doesn't get spammy.
+
+	var/alarm_delay = 2000
+
+/obj/machinery/metal_detector/New()
+	..()
+	next_announcement_time = get_game_time() - alarm_delay
+
+	component_parts = list()
```

Comment:
This entire section should use `default_apply_parts()`, instead of defining components individually. The proc is found in machinery.dm line 308.

Make a circuitboard:
`/obj/item/weapon/circuitboard/metaldetector`, and place it in `code/game/objects/items/weapons/circuitboards/machinery/metaldetectors.dm`

Define the name, built_path as the metaldetector, the board_type as a frame (you make the frame in `code/game/machinery/frame.dm` and path it `/datum/frame/frame_types/metaldetector`), you define the matter as whatever you prefer for making boards, default is wallmaterial = 50, and req_components is that components list you had originally.

For reference, said components list would look like:
`req_components = list(
							/obj/item/weapon/stock_parts/console_screen = 1,
							/obj/item/weapon/stock_parts/scanning_module = 3)`
Circuitboard not included because we define it earlier on the object.

Then, to enable it being printable, go to:
`code/modules/research/designs/circuits/circuits.dm` and add a build path:
`/datum/design/circuit/metaldetector` and set the build_path to the circuitboard we defined earlier.

Then, in metaldetector.dm, set `circuit = /obj/item/weapon/circuitboard/metaldetector`

And that's it, your items are now createable in-game, using the correct method for generating components!
You'll also want to remove `RefreshParts()` on the init because default_apply_parts() does that for you.